plugins/GSdx/GSRendererOGL.cpp

```diff
@@ -234,7 +234,14 @@ void GSRendererOGL::EmulateTextureShuffleAndFbmask()
 		// Avious dubious call to m_texture_shuffle on 16 bits games
 		// The pattern is severals column of 8 pixels. A single sprite
 		// smell fishy but a big sprite is wrong.
-		m_texture_shuffle = ((v[1].U - v[0].U) < 256);
+
+		// Tomb Raider Angel of Darkness relies on this behavior to prodce a fog effect.
+		// In this case, the address of the framebuffer 
+		// and texture are the same. 
+		// The game will take RG => BA and then the BA => RG of next pixels. 
+		// However, only RG => BA needs to be emulated because RG isn't used.
+		GL_INS("WARNING: Possible misdetection of a texture shuffle effect");
+		m_texture_shuffle = ((v[1].U - v[0].U) < 256) || ((m_context->FRAME.Block() == m_context->TEX0.TBP0));
```

Comment:
There are double brackets around m_context->FRAME.Block() == m_context->TEX0.TBP0 ;)

Reply:
It was nearly perfect. I'll fix it after I eat :P

Reply:
I think all the brackets (except after Block of course) are unnecessary? Though perhaps the brackets around `v[1].U - v[0].U` helps improve clarity.

Reply:
I assume you want me to remove them?

Reply:
Personally I would, but I don't know if others will have trouble figuring out the evaluation order.